generated/nirfsg/nirfsg/session.py

```diff
@@ -52,43 +52,1095 @@ def __exit__(self, exc_type, exc_value, traceback):
+
+                    **Default Value:** MarkerEventToggleInitialState.LOW
+
+                    **Supported Devices:** PXIe-5820/5830/5831/5832/5840/5841/5842
+
+                    **Related Topics**
+
+                    `Marker Events <https://www.ni.com/docs/en-US/bundle/rfsg/page/rfsg/marker_events.html>`_
+
+                **Defined Values**:
+
+    +------------------------------------+----------------+----------------------------------------------------------------------------------+
+    | Name                               | Value          | Description                                                                      |
+    +====================================+================+==================================================================================+
+    | MarkerEventToggleInitialState.HIGH | 21001 (0x5209) | Specifies the initial state of the Marker Event toggle behavior as digital high. |
+    +------------------------------------+----------------+----------------------------------------------------------------------------------+
+    | MarkerEventToggleInitialState.LOW  | 21000 (0x5208) | Specifies the initial state of the Marker Event toggle behavior as digital low.  |
+    +------------------------------------+----------------+----------------------------------------------------------------------------------+
+
+    Tip:
+    This property can be set/get on specific markers within your :py:class:`nirfsg.Session` instance.
+    Use Python index notation on the repeated capabilities container markers to specify a subset.
+
+    Example: :py:attr:`my_session.markers[ ... ].marker_event_toggle_initial_state`
+
+    To set/get on all markers, you can call the property directly on the :py:class:`nirfsg.Session`.
+
+    Example: :py:attr:`my_session.marker_event_toggle_initial_state`
+    '''
+    def __init__(self, session, repeated_capability_list):
+        object.__setattr__(self, '_session', session)
+        object.__setattr__(self, '_repeated_capability_list', repeated_capability_list)
+        object.__setattr__(self, '_prefix', 'marker')
+        object.__setattr__(self, '_current_repeated_capability_list', repeated_capability_list if len(repeated_capability_list) > 0 else [''])
+        object.__setattr__(self, '_separator', '')
+
+    def __setattr__(self, key, value):
+        if key not in dir(self):
+            raise AttributeError("'{0}' object has no attribute '{1}'".format(type(self).__name__, key))
+        object.__setattr__(self, key, value)
+
+    def __getitem__(self, repeated_capability):
+        '''Set/get properties or call methods with a repeated capability (i.e. channels)'''
+        rep_caps_list = _converters.convert_repeated_capabilities(repeated_capability, self._prefix)
+        complete_rep_cap_list = [
+            current_rep_cap + self._separator + rep_cap
+            for current_rep_cap in self._current_repeated_capability_list
+            for rep_cap in rep_caps_list
+        ]
```

Comment:
This list comprehension exists to support the chaining of repeated capabilities, i.e. `session.sites[0, 1].pins['PinA', 'PinB'].ppmu_source()`.
See https://github.com/ni/nimi-python/pull/1291

This new implementation won't work with chaining.
You can't call .pins in a class where we don't define properties/attributes for each of the other rep cap classes.